src/VisualStudio/IntegrationTest/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -506,5 +574,181 @@ private T Retry<T>(Func<T> action, Func<T, bool> stoppingCondition, TimeSpan del
+                    {
+                        waitHandle.Set();
+                    }
+                }
+
+                componentChangeService.ComponentChanged += ComponentChanged;
+
+                try
+                {
+                    InvokeOnUIThread(() =>
+                    {
+                        var button = designerHost.Container.Components[buttonName];
+                        var properties = TypeDescriptor.GetProperties(button);
+                        var property = properties[propertyName];
+                        if (propertyTypeName == null)
+                        {
+                            property.SetValue(button, propertyValue);
+                        }
+                        else
+                        {
+                            var enumPropertyValue = GetEnumPropertyValue(propertyTypeName, propertyValue);
+                            property.SetValue(button, enumPropertyValue);
+                        }
+                    });
+                    waitHandle.WaitOne();
+                }
+                finally
+                {
+                    componentChangeService.ComponentChanged -= ComponentChanged;
+                }
+            }
+        }
+
+        public void EditWinFormButtonEvent(string buttonName, string eventName, string eventHandlerName)
+        {
+            using (var waitHandle = new ManualResetEvent(false))
+            {
+                var designerHost = (IDesignerHost)GetDTE().ActiveWindow.Object;
+                var componentChangeService = (IComponentChangeService)designerHost;
+                void ComponentChanged(object sender, ComponentChangedEventArgs e)
+                {
+                    if (e.Member.Name == eventName)
+                    {
+                        waitHandle.Set();
+                    }
+                }
+
+                componentChangeService.ComponentChanged += ComponentChanged;
+
+                try
+                {
+                    InvokeOnUIThread(() =>
+                    {
+                        var button = designerHost.Container.Components[buttonName];
+                        var eventBindingService = (IEventBindingService)button.Site.GetService(typeof(IEventBindingService));
+                        var events = TypeDescriptor.GetEvents(button);
+                        var eventProperty = eventBindingService.GetEventProperty(events.Find(eventName, ignoreCase: true));
+                        eventProperty.SetValue(button, eventHandlerName);
+                    });
+                    waitHandle.WaitOne();
+                }
+                finally
+                {
+                    componentChangeService.ComponentChanged -= ComponentChanged;
+                }
+            }
+        }
+
+        public void VerifyWinFormButtonPropertySet(string buttonName, string propertyName, string expectedPropertyValue)
```

Comment:
Keep verification code in the tests themselves (or in the test class as a helper method). The InProc/OutOfProc components should only be concerned with automating VS and returning information about its state.

In this particular case I'd rather see a `GetWinFormButtonProperty` method that simply returned the property value (or perhaps null if it isn't set; or maybe some sort of `Optional` data type to indicate whether or not it is set). The verification code on the test side could use actual xUnit assertions.